chain/src/view.rs

```diff
@@ -112,11 +113,18 @@ pub trait StateReadExt: StateRead {
         // Get the height
         let height = self.get_block_height().await?;
 
-        // Get the epoch duration
-        let epoch_duration = self.get_epoch_duration().await?;
+        self.get(&state_key::epoch_by_height(height))
+            .await?
+            .ok_or_else(|| anyhow!("missing epoch for current height"))
+    }
 
-        // The current epoch
-        Ok(Epoch::from_height(height, epoch_duration))
+    /// Gets the pending epoch change, if any
+    async fn pending_epoch(&self) -> Result<Option<Epoch>> {
+        Ok(self
+            .get(&state_key::epoch_change_at_height(
+                self.get_block_height().await?,
+            ))
+            .await?)
```

Comment:
What's the meaning of a pending epoch change?

Reply:
This was meant to convey to begin_block that it should increment the epoch, but I changed this so that the epoch is set by the app for the next block every time in `end_block instead, so this is no longer needed